spec/lib/kiev/shoryuken_spec.rb

```diff
@@ -0,0 +1,174 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+require "kiev/test"
+
+if defined?(::Shoryuken)
+  RSpec.describe Kiev::Shoryuken do
+    # {
+    #   message_id: "437eb14c-7a98-457c-a431-01671564237e",
+    #   body: "test",
+    #   message_attributes: message_attributes,
+    #   delete: nil
+    # }
+    # {
+    #   "request_id"    => "acc6acfe-525e-49a6-a12b-ce3f07564620",
+    #   "request_depth" => 1,
+    #   "tree_path"     => "B"
+    # }
+
+    before(:all) do
+      # Shoryuken skips configure_server blocks unless it's running in a worker
+      # process. "worker process"ness is defined only by existence of this module
+      module Shoryuken::CLI
+      end
+
+      Kiev::Shoryuken.enable
+      Kiev::Test::Log.configure
+    end
+
+    describe "client middleware when sending a message" do
```

Comment:
I would suggest to write tests for each middleware separately and also test for `Kiev::Shoryuken.enable` method, but with a simple check that all necessary middlewares were added into the chain. I believe it will be simple than attempt to cover whole bunch of things here.

Reply:
Think of that as functional tests, I guess :)
Unit tests would be of use, I guess.